switchapi/es2k/switch_fdb.c

```diff
@@ -291,9 +291,19 @@ switch_status_t switch_api_l2_forward_create(const switch_device_t device,
 
       status = switch_rif_get(device, api_l2_info->rif_handle, &rif_info);
       CHECK_RET(status != SWITCH_STATUS_SUCCESS, status);
+
+      /* Get appropiate port_id if not already set */
+      switch_pd_to_get_port_id(&rif_info->api_rif_info);
```

Comment:
`switch_pd_to_get_port_id()` has two failure cases. Shouldn't it return a status code?

Reply:
in switch_pd_to_get_port_id(), error when returning "rmac info" not found or "rmac_entry" not found if fine. There are cases where valid rmac wouldnt have been created when we call this API.